File: src/GroundZero.Application/Features/Reviews/DTOs/CreateReviewRequest.cs
using GroundZero.Domain.Enums;

namespace GroundZero.Application.Features.Reviews.DTOs;

public class CreateReviewRequest
{
    public int Rating { get; set; }
    public string? Comment { get; set; }
    public ReviewType ReviewType { get; set; }
    public int? ProductId { get; set; }
    public int? AppointmentId { get; set; }
}

File: groundzero_mobile/lib/features/shop/models/create_review_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'create_review_request.freezed.dart';
part 'create_review_request.g.dart';

@freezed
class CreateReviewRequest with _$CreateReviewRequest {
  const factory CreateReviewRequest({
    required int rating,
    String? comment,
    required int reviewType, // 0 = Product, 1 = Appointment
    int? productId,
    int? appointmentId,
  }) = _CreateReviewRequest;

  factory CreateReviewRequest.fromJson(Map<String, dynamic> json) =>
      _$CreateReviewRequestFromJson(json);
}
